services/wallet/reader.go

```diff
@@ -264,7 +264,10 @@ func (r *Reader) invalidateBalanceCache() {
 func (r *Reader) FetchOrGetCachedWalletBalances(ctx context.Context, clients map[uint64]chain.ClientInterface, addresses []common.Address) (map[common.Address][]token.StorageToken, error) {
 	needFetch := !r.isBalanceCacheValid(addresses) || r.isBalanceUpdateNeededAnyway(clients, addresses)
 	if needFetch {
-		return r.FetchBalances(ctx, clients, addresses)
```

Comment:
@belalshehab I am not really sure how does this line differ from what you added in those 3 lines below? Isn't the effect the same?

Reply:
here I want to only return fetched balances if no error, otherwise `return r.GetCachedBalances(clients, addresses)` is executed

Reply:
As long as you get an error from `FetchBalances` `fetchedBalances` will be `nil`, when an error is `nil`, `fetchedBalances` will be set. That's why I said the end effect is the same. So feel free to ever this change.

Reply:
but if I do `return r.FetchBalances(ctx, clients, addresses)` then I will not get the cachedBalances in case of error
the client will get error when fetching data, but last known data will not be shown

Reply:
I am not sure how?
Could you tell me what's exactly the difference between:
`return r.FetchBalances(ctx, clients, addresses)`
and
```
fetchedBalances, err := r.FetchBalances(ctx, clients, addresses)
if err == nil {
	return fetchedBalances, nil
}
```

Reply:
Ohhh sorry, I missed that, I see the point now.